packages/cms/pages/settings/roles-and-permissions/roles-and-permissions.tsx

```diff
@@ -379,11 +415,36 @@ const RolesTable: React.FC = () => {
     {
       field: 'members',
       name: 'Members'
+    },
+    {
+      field: 'actions',
+      name: 'Actions',
+      actions: [
+        {
+          name: 'Edit',
+          description: 'Edit role',
+          icon: 'pencil',
+          type: 'icon',
+          onClick: (role: any) => {
+            form.setForm({
+              id: role?.id,
+              name: role?.name,
+              description: role?.description,
+              slug: role?.slug,
+              adminPermissions: getPermissionIDs(role?.adminPermissions)
+            })
+            setIsEditing(true)
```

Comment:
We can actually remove this state and check if `form.id` then its editing, if not then its creating. That way we have much less state.

Reply:
okay